packages/mesa_amber.rb

```diff
@@ -0,0 +1,270 @@
+
+      -   if(LLVMIsConstant(a) && LLVMIsConstant(b)) {
+      -      if (type.floating)
+      -         res = LLVMConstFMul(a, b);
+      -      else
+      -         res = LLVMConstMul(a, b);
+      -      if(shift) {
+      -         if(type.sign)
+      -            res = LLVMConstAShr(res, shift);
+      -         else
+      -            res = LLVMConstLShr(res, shift);
+      -      }
+      -   }
+      -   else {
+      -      if (type.floating)
+      -         res = LLVMBuildFMul(builder, a, b, "");
+      +   if (type.floating)
+      +       res = LLVMBuildFMul(builder, a, b, "");
+      +    else
+      +       res = LLVMBuildMul(builder, a, b, "");
+      +    if (shift) {
+      +       if (type.sign)
+      +          res = LLVMBuildAShr(builder, res, shift, "");
+             else
+      -         res = LLVMBuildMul(builder, a, b, "");
+      -      if(shift) {
+      -         if(type.sign)
+      -            res = LLVMBuildAShr(builder, res, shift, "");
+      -         else
+      -            res = LLVMBuildLShr(builder, res, shift, "");
+      -      }
+      +          res = LLVMBuildLShr(builder, res, shift, "");
+          }
+
+          return res;
+      @@ -1288,15 +1256,6 @@ lp_build_div(struct lp_build_context *bl
+          if(a == bld->undef || b == bld->undef)
+             return bld->undef;
+
+      -   if(LLVMIsConstant(a) && LLVMIsConstant(b)) {
+      -      if (type.floating)
+      -         return LLVMConstFDiv(a, b);
+      -      else if (type.sign)
+      -         return LLVMConstSDiv(a, b);
+      -      else
+      -         return LLVMConstUDiv(a, b);
+      -   }
+      -
+          /* fast rcp is disabled (just uses div), so makes no sense to try that */
+          if(FALSE &&
+             ((util_get_cpu_caps()->has_sse && type.width == 32 && type.length == 4) ||
+      @@ -2643,7 +2602,7 @@ lp_build_rcp(struct lp_build_context *bl
+          assert(type.floating);
+
+          if(LLVMIsConstant(a))
+      -      return LLVMConstFDiv(bld->one, a);
+      +      return LLVMBuildFDiv(builder, bld->one, a, "");
+
+          /*
+           * We don't use RCPPS because:
+
+    PATCH_EOF
+
+    # llvm 13/14 patch  See https://gitlab.freedesktop.org/mesa/mesa/-/issues/5455
+    # & https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/13273.patch
+    system 'patch', '-Np1', '-i', '13273.diff'
+    # mesa: Implement ANGLE_sync_control_rate (used by Chrome browser)
+    system 'patch', '-Np1', '-i', '15381.diff'
+    # llvm 15 patch
+    system 'patch', '-Np1', '-i', '15091.diff'
+    system 'patch', '-Np1', '-i', '15232.diff'
+    system 'patch', '-Np1', '-i', '16129.diff'
+    system 'patch', '-Np1', '-F', '10', '-i', '16289.diff'
+    system 'patch', '-Np1', '-i', '17514.diff'
+    system 'patch', '-Np1', '-i', '17518.patch'
+  end
+
+  def self.build
+    system <<~BUILD
+      mold -run meson setup #{CREW_MESON_OPTIONS} \
+        -Db_asneeded=false \
+        -Damber=true \
```

Comment:
If I recall correctly setting `-Damber=true` kept certain drivers from being built, which kept sommelier from working. Do you see that at all?